python-backend/app/api/search/search/resources/simple_search.py

```diff
@@ -0,0 +1,94 @@
+
+        reg_exp = regex.compile(r'\'.*?\' | ".*?" | \S+ ', regex.VERBOSE)
+        search_terms = reg_exp.findall(search_term)
+        search_terms = [term.replace('"', '') for term in search_terms]
+
+        with ThreadPoolExecutor(max_workers=50) as executor:
+            task_list = []
+            for type, type_config in simple_search_targets.items():
+                task_list.append(
+                    executor.submit(execute_search, app, search_results, search_term, search_terms,
+                                    type, type_config))
+            for task in as_completed(task_list):
+                try:
+                    data = task.result()
+                except Exception as exc:
+                    current_app.logger.error(
+                        f'generated an exception: {exc} with search term - {search_term}')
+
+        grouped_results = {}
+        for result in search_results:
+            if (result.result['id'] in grouped_results):
+                grouped_results[result.result['id']].score += result.score
+            else:
+                grouped_results[result.result['id']] = result
+
+        search_results = list(grouped_results.values())
+        search_results.sort(key=lambda x: x.score, reverse=True)
+        search_results = search_results[0:10]
+
+        return {'search_terms': search_terms, 'search_results': search_results}
+
+
+def append_result(search_results, search_term, type, item, id_field, value_field, score_multiplier):
+
+    if getattr(item, value_field).lower().startswith(search_term.lower()):
+        score_multiplier = score_multiplier * 3
+
+    if getattr(item, value_field).lower() == search_term.lower():
+        score_multiplier = score_multiplier * 10
+
+    search_results.append(
+        SearchResult(
+            getattr(item, 'score') * score_multiplier, type, {
+                'id': getattr(item, id_field),
+                'value': getattr(item, value_field)
+            }))
+
+
+def execute_search(app, search_results, search_term, search_terms, type, type_config):
+    with app.app_context():
+        for term in search_terms:
+            if len(term) > 2:
+                for column in type_config['columns']:
+                    similarity = db.session.query(Mine).with_entities(
+                        func.similarity(column, term).label('score'),
+                        *type_config['entities']).filter(column.ilike(f'%{term}%'))
+                    if type_config['has_deleted_ind']:
+                        similarity = similarity.filter_by(deleted_ind=False)
+                    similarity = similarity.order_by(desc(func.similarity(column, term))).all()
+                    [
+                        append_result(search_results, search_term, type, item,
+                                      type_config['id_field'], type_config['value_field'],
+                                      type_config['score_multiplier']) for item in similarity
+                    ]
```

Comment:
same execute_search and append_result?

Reply:
possibly move simple search, search, and search options into the same file, to share these methods.

Reply:
Moved these to the helper class.